nucleus/core/logging/osgi.bundle

```diff
@@ -41,13 +41,20 @@
 # logviewer.backend is used by REST implementation. We should be able to remove that when we move
 # REST logging code to this module.
 # logging and diagnostics are used by amx!
--exportcontents: com.sun.enterprise.server.logging.logviewer.backend, com.sun.enterprise.server.logging.diagnostics, com.sun.enterprise.server.logging
+-exportcontents: \
+          com.sun.enterprise.server.logging.logviewer.backend, \
+          com.sun.enterprise.server.logging.diagnostics, \
+          com.sun.enterprise.server.logging
 
-DynamicImport-Package: \
-					org.glassfish.cluster.ssh.launcher, \
-                    org.glassfish.cluster.ssh.sftp, \
-					com.trilead.ssh2, \
-					com.sun.enterprise.util.cluster.windows.io, \
-					com.sun.enterprise.util.cluster.windows.process, \
-					org.glassfish.cluster.ssh.util
+Import-Package: \
+          fish.payara.logging.jul.*; resolution:=optional, \
```

Comment:
As always, I don't believe this is optional dependency.

Reply:
I tried to find other way, but this is the only one working:
PayaraLogManager must be on boot classpath, otherwise it is not used, because JVM uses it's own first. Server's classloader hierarchy is bit overcomplicated, that is why I created that custom classloader printing list of jar files it knows. It is useful also with verbose logging or debugging, simply I always see on the first sight which classloader is used right now and where.
But back to osgi. When I make it mandatory, Felix tries to find it only in classloaders managed by him for some reason. With this and the change in osgi.properties it allows to acces packages from bootstrap (like other from this "level").

Maybe I am missing something, but then I need a help.

Reply:
Yes, there is a way, I'll look it up.